packages/headless/src/app/search-engine/search-engine.ssr.test.ts

```diff
@@ -149,6 +160,23 @@ describe('SSR', () => {
       expect(getResultsPerPage(staticState)).toBe(defaultNumberOfResults);
     });
 
+    it('should call augmentPreprocessRequestWithForwardedFor when fetchStaticState is invoked', async () => {
+      const spy = vi.spyOn(
+        augmentModule,
+        'augmentPreprocessRequestWithForwardedFor'
+      );
+
+      const fetchStaticState = engineDefinition.fetchStaticState;
+      await fetchStaticState();
+      expect(spy).toHaveBeenCalledWith({
+        loggerOptions: {level: 'warn'},
+        navigatorContextProvider: mockNavigatorContextProvider,
+        preprocessRequest: mockPreprocessRequest,
+      });
+
+      spy.mockRestore();
```

Comment:
Just an idea - and certainly a bit out of scope - but this makes me realize we could instead add [`restoreMocks: true`](https://vitest.dev/config/#restoremocks) in the headless vitest.config.js.

This might require some work, though, as some tests may actually rely on the fact that we're not restoring mocks between tests. But we won't know if we don't try it.